### ISSUE TITLE
FLTK linking error

### ISSUE THREAD

**Issue by cenit** [Microsoft/vcpkg#1243]
Windows 10 CU, latest vcpkg, FLTK just installed as standard triplet (x86-windows), VS 15.2 (latest release, everything fully updated)

This is a very simple (I hope) test case that does not work anymore:

```
#include <stdio.h>
#include <FL/Fl.H>
#include <FL/Fl_Window.H>
#include <FL/Fl_Box.H>

class MyBox : public Fl_Box {
public:
    int handle(int e) {
        if ( e == FL_PUSH ) {
            fprintf(stdout, "event_x,event_y: %d,%d\n",
                Fl::event_x(),
                Fl::event_y()
            );
        }
        return Fl_Box::handle(e);
    }

    MyBox(int x,int y,int w,int h,const char*l=0) : Fl_Box(x,y,w,h,l) {
        color(FL_BLUE);
        box(FL_FLAT_BOX);
    }
};


int main() {
    Fl_Window win(800, 600);
    MyBox box(0,0,800,600);
    win.end();

    win.show();
    Fl::run();

    return 0;
} 
```

It used to work in the past, after updating all outdated libs (which included FLTK) now it is broken.
Still works under every other supported toolchain. Looks like some symbols (if I add other events in the code the list can increase) are not to be found in the vcpkg produced libraries.

The specific error in this case is:
```
error LNK2001: unresolved external symbol "public: static int Fl::e_x" (?e_x@Fl@@2HA)
error LNK2001: unresolved external symbol "public: static int Fl::e_y" (?e_y@Fl@@2HA)
```
Any help is appreciated. I'm just moving my team to vcpkg but these problems are a big trouble when trying to convince to use a new tool (which is great IMHO and which I'm using since November!)

**Comment by codicodi** [comment]
Looking at mangled symbols I suspect you're missing some kind of import macro.

**Comment by codicodi** [comment]
Try adding `/DFL_DLL` to your compiler flags

**Comment by codicodi** [comment]
...and that was a temporary solution. See here for a more robust one: #1247

**Comment by cenit** [comment]
thank you so much! The temporary solution works well, just waiting for your pr to be accepted to remove it!
How did you manage to find this trick?

**Comment by codicodi** [comment]
Happy to help :)
Many libraries have some macro to control it's linkage (shared or static) and rely on users to define it (or not) depending on what type of library they link to.
Note that while #1247 solves this problem for vcpkg users, you would still need to take care of this macro when linking to fltk DLLs not provided by vcpkg.

**Comment by ras0219-msft** [comment]
Merged #1247! Happy coding!
